language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentService.java

```diff
@@ -289,14 +296,104 @@ public CompletableFuture<BallerinaSyntaxTreeResponse> triggerModify(BallerinaTri
         return CompletableFuture.supplyAsync(() -> reply);
     }
 
+    @JsonRequest
+    public CompletableFuture<BallerinaSyntaxTreeResponse> syntaxTreeByName(BallerinaSyntaxTreeByNameRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            AtomicReference<String> sourceCode = new AtomicReference<>();
```

Comment:
IIRC here we won't need to use Atomic References as these are in-scope variables of the `supplyAsync` invocation.